cloudwatch_logs_common/src/utils/log_file_manager.cpp

```diff
@@ -50,11 +63,45 @@ FileObject<LogCollection> LogFileManager::readBatch(
     Aws::String aws_line(line.c_str());
     Aws::Utils::Json::JsonValue value(aws_line);
     Aws::CloudWatchLogs::Model::InputLogEvent input_event(value);
-    actual_batch_size++;
-    log_set.insert(input_event);
-    data_tokens.push_back(data_token);
+    pq.push(std::make_tuple(input_event.GetTimestamp(), line, data_token));
   }
-  LogCollection log_data(log_set.begin(), log_set.end());
+  
+  /*  
+    Set latest timestamp to top of pq
+    Go through the entire pq
+    Only store logs with < 24 hours diff from latest time
+    Count these logs as actual_batch_size
+    The older logs will be uploaded on the next readBatch command
+  */
+
+  Timestamp latestTime = std::get<0>(pq.top());
+  LogCollection log_data;
+  std::list<FileManagement::DataToken> data_tokens;
+  size_t actual_batch_size = 0;
```

Comment:
Why not use [std::list::size()](https://en.cppreference.com/w/cpp/container/list/size)?